doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -126,42 +133,72 @@ Type declarations
 
 One can define new types with the following commands.
 
-.. cmd:: Ltac2 Type {? @ltac2_typeparams } @lident
+.. cmd:: Ltac2 Type {? rec } @tac2typ_def {* with @tac2typ_def }
    :name: Ltac2 Type
 
-   This command defines an abstract type. It has no use for the end user and
-   is dedicated to types representing data coming from the OCaml world.
-
-.. cmdv:: Ltac2 Type {? rec} {? @ltac2_typeparams } @lident := @ltac2_typedef
-
-   This command defines a type with a manifest. There are four possible
-   kinds of such definitions: alias, variant, record and open variant types.
-
-   .. productionlist:: coq
-      ltac2_typedef    : `ltac2_type`
-                       : [ `ltac2_constructordef` | ... | `ltac2_constructordef` ]
-                       : { `ltac2_fielddef` ; ... ; `ltac2_fielddef` }
-                       : [ .. ]
-      ltac2_constructordef   : `uident` [ ( `ltac2_type` , ... , `ltac2_type` ) ]
-      ltac2_fielddef         : [ mutable ] `ident` : `ltac2_type`
-
-   Aliases are just a name for a given type expression and are transparently
+   .. insertprodn tac2typ_def tac2rec_field
+
+   .. prodn::
+      tac2typ_def ::= {? @tac2typ_prm } @qualid {? {| := | ::= } @tac2typ_knd }
+      tac2typ_prm ::= @typ_param
+      | ( {+, @typ_param } )
+      tac2typ_knd ::= @tac2type5
+      | [ {? {? %| } {+| @tac2alg_constructor } } ]
+      | [ .. ]
+      | %{ {? {+; @tac2rec_field } {? ; } } %}
+      tac2alg_constructor ::= @ident
+      | @ident ( {*, @tac2type5 } )
+      tac2rec_field ::= {? mutable } @ident : @tac2type5
+
+   :n:`:=`
+     Defines a type with a manifest
```

Comment:
I think it's the official nomenclature in OCaml, at least in the implementation.

Reply:
I think "manifest" definitely needs to be defined here.  Maybe "defines a type with an explicit set of constructors"?  It may be official in the OCaml world, but it's not standard in the Coq world.